### PR TITLE
Fix pType parser.

### PR DESCRIPTION
**Pull request by thealmarty**
As @michaelpj suggested. And @effectfully also pointed out this *may* fix this https://hydra.iohk.io/build/12938326/nixlog/68.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-core/plutus-core/src/PlutusCore/Parser/ParserCommon.hs [hunk 1]
```diff
@@ -113,8 +113,7 @@ kind = inParens (typeKind <|> funKind)
 -- | Parser for @PType@.
 pType :: Parser PType
 pType = choice
```
Comment (michaelpj) [inline plutus-core/plutus-core/src/PlutusCore/Parser/ParserCommon.hs:115]: you're missing the `map try`

**Comment by thealmarty** [comment]
Just realized I'll have to add a test to test this particularly because the failed test there is generated.

**Comment by michaelpj** [comment]
I'm not too worried about adding a test for this particular thing, as long as the existing tests keep passing I'm happy. It's just obviously more right since it's in line with all the other parsers.

**Review by kwxm — COMMENTED** [review]
Looks OK.

**Pull request merged**